### PR TITLE
Governance / CIP-95

### PR DESCRIPTION
**Pull request by errfrom**
~~Depends on https://github.com/mlabs-haskell/cardano.nix/pull/96~~ merged
Depends on https://github.com/mlabs-haskell/purescript-cardano-hd-wallet/pull/3
Depends on https://github.com/mlabs-haskell/purescript-cardano-key-wallet/pull/3
Depends on https://github.com/mlabs-haskell/purescript-cardano-transaction-builder/pull/3
Depends on https://github.com/mlabs-haskell/purescript-cip30-mock/pull/2
Depends on https://github.com/mlabs-haskell/purescript-cip95/pull/2

### Governance examples (Ogmios + Kupo, SanchoNet, Eternl Beta & KeyWallet):
- `Gov.ManageDrep`
  - Covered certificates:
    - `RegDrepCert`
    - `UpdateDrepCert`
    - `UnregDrepCert`
- `Gov.ManageDrepScript`
- `Gov.DelegateVoteAbstain`
  - Covered certificates:
    - `VoteRegDelegCert`
- `Gov.SubmitVote`
- `Gov.SubmitVoteScript`
    
### New CIP-95 functions exported from `Contract.Wallet`:

- `ownDrepPubKey`
- `ownDrepPubKeyHash`
- `ownRegisteredPubStakeKeys`
- `ownUnregisteredPubStakeKeys`

**NOTE**: KeyWallet does not distinguish between registered and unregistered stake keys due to the limitations of the underlying query layer. This means that all controlled stake keys are returned as part of `ownUnregisteredPubStakeKeys`, and the response of `ownRegisteredPubStakeKeys` is always an empty array.

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior are covered with tests
- [x] The template (`templates/ctl-scaffold`) has been [updated](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/development.md#maintaining-the-template)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Changed`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: test/Ogmios/Aeson.purs [hunk 1]
```diff
@@ -127,16 +127,19 @@ loadFixtures = do
...
-        (error <<< ((bn <> "\n  ") <> _) <<< printJsonDecodeError)
-        (Aeson.parseJsonStringToAeson contents)
-      pure case pattern >>= flip match bn >>> map tail of
-        Just [ Just query ] -> Just
-          { query
-          , bn
-          , aeson
-          }
-        _ -> Nothing
+      -- ignore empty (corrupted) fixtures
```
Comment (klntsky) [inline test/Ogmios/Aeson.purs:130]: Are there any? Where are they from? can we get rid of them?

File: src/Internal/Wallet/Cip30.purs [hunk 2]
```diff
@@ -79,13 +84,15 @@ type Cip30Wallet =
   , getRewardAddresses :: Aff (Array Address)
   , signTx :: Transaction -> Aff Transaction
   , signData :: Address -> RawBytes -> Aff DataSignature
+  , getPubDrepKey :: Aff PublicKey
```
Comment (klntsky) [inline src/Internal/Wallet/Cip30.purs:87]: Is it always available? Shouldn't it be `Maybe PublicKey`?

File: test/Ogmios/Aeson.purs (hunk 1 above)
Comment (errfrom) [reply test/Ogmios/Aeson.purs:130]: Yes, there is at least one in the ogmios repository

File: src/Internal/Wallet/Cip30.purs (hunk 2 above)
Comment (errfrom) [reply src/Internal/Wallet/Cip30.purs:87]: This key is derived from the account, so I'd assume it should be always available

Comment (errfrom) [reply src/Internal/Wallet/Cip30.purs:87]: At least I don't see any notes in the CIP-95 about it being optional, let's revisit this concern later

File: src/Internal/Contract/Wallet.purs [hunk 3]
```diff
@@ -251,3 +279,51 @@ getWalletUtxos = do
...
+ownUnregisteredPubStakeKeys :: Contract (Array PublicKey)
+ownUnregisteredPubStakeKeys =
+  withWallet do
+    actionBasedOnWallet _.getUnregisteredPubStakeKeys
+      (kwPubStakeKeys "ownUnregisteredPubStakeKeys")
+
+kwPubStakeKeys :: String -> KeyWallet -> Contract (Array PublicKey)
+kwPubStakeKeys funName kw = do
+  logWarn' $ funName <>
+    " via KeyWallet: KeyWallet does not distinguish between \
```
Comment (klntsky) [inline src/Internal/Contract/Wallet.purs:321]: I think the spec implies that if it's impossible to know, all keys should be considered unregistered:

> api.cip95.getUnregisteredPubStakeKeys
> If the wallet does not know the registration status of it's stake keys then it should return them as part of this call

File: src/Internal/Wallet/Cip30.purs (hunk 2 above)
Comment (klntsky) [reply src/Internal/Wallet/Cip30.purs:91]: OK let's consider it always available

File: src/Internal/Contract/Wallet.purs (hunk 3 above)
Comment (errfrom) [reply src/Internal/Contract/Wallet.purs:321]: That's a great point, I'll update the implementation

Comment (errfrom) [reply src/Internal/Contract/Wallet.purs:330]: Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1638/commits/cbf1fa857e45d1a8717ddfccb5ebe8df82f36465

File: flake.nix [hunk 4]
```diff
@@ -172,9 +172,6 @@
                 nodePackages.eslint
                 nodePackages.prettier
                 blockfrost-backend-ryo
-                cardano-node.packages.${system}.cardano-testnet
-                cardano-node.packages.${system}.cardano-cli
```
Comment (klntsky) [inline flake.nix:176]: Why?

File: nix/test-nixos-configuration.nix [hunk 5]
```diff
@@ -35,8 +35,8 @@
     hostAddr = "0.0.0.0";
     socketPath = "/var/run/cardano-node/node.socket";
     systemdSocketActivation = false;
-    nodeConfigFile = "${cardano-configurations}/network/preview/cardano-node/config.json";
-    topology = "${cardano-configurations}/network/preview/cardano-node/topology.json";
+    nodeConfigFile = "${cardano-configurations}/network/sanchonet/cardano-node/config.json";
```
Comment (klntsky) [inline nix/test-nixos-configuration.nix:38]: We need to switch back to preview

Comment (errfrom) [reply nix/test-nixos-configuration.nix]: Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1638/commits/5a745da036583337d658b27a4dd96dcfd5ce3699

File: flake.nix (hunk 4 above)
Comment (errfrom) [reply flake.nix:176]: Because these packages are provided as part of the runtime (`withRuntime` flag)

File: packages.dhall [hunk 6]
```diff
@@ -220,9 +254,14 @@ let additions =
...
-          [ "aff-promise", "console", "effect", "functions", "prelude" ]
+          [ "aff-promise"
+          , "console"
+          , "effect"
+          , "functions"
+          , "prelude"
+          ]
         , repo = "https://github.com/mlabs-haskell/purescript-cip30-mock"
-        , version = "v1.0.0"
+        , version = "d00ba14769dceda1270658c63f14822db01f376d"
```
Comment (klntsky) [inline packages.dhall:264]: Please release new versions for every package and use them here

**Review by klntsky — CHANGES_REQUESTED** [review]

**Review by klntsky — APPROVED** [review]

**Pull request merged**